### DISCUSSION TITLE
[Recommendation] Floorp might have too many, unmaintainable options

### DISCUSSION THREAD

**Discussion by jacksongoode** [Floorp-Projects/Floorp#509] [Feature Request]
As a general observation, not every theme option was compatible with one another. I would either consolidate to the most popular theme options or make every aspect of theming (color, tab style, padding, hiding of address/tab bar) modular so that they can be controlled.

It's quite confusing to see multiple themes, each with their own options in addition to options that override some of those theme settings. None of the options should look awkward or glitch the browser, but many do.

I can see this buckling under the weight of upstream FF changes.

**Comment by Ryosuke-Asano** [comment]
You are absolutely right.

I have separated the Firefox source code from the Floorp source code for that reason.

You can get the Floorp source code from /floorp-core.

I also avoid compatibility problems with CSS by rewriting almost everything in JS.

ESModule is also actively used, so upstream problems can be quickly resolved.

**Comment by Ryosuke-Asano** [comment]
BTW, I create new browser resole the problem

https://www.reddit.com/r/Floorp/comments/16xhw7h/introduce_floorp_lightning/
